dask_image/feature/template.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import dask.array as da
+from dask_image import ndfilters
+
+
+def _window_sum_2d(image, window_shape):
+
+    window_sum = da.cumsum(image, axis=0)
+    window_sum = (window_sum[window_shape[0]:-1]
+                  - window_sum[:-window_shape[0] - 1])
+
+    window_sum = da.cumsum(window_sum, axis=1)
+    window_sum = (window_sum[:, window_shape[1]:-1]
+                  - window_sum[:, :-window_shape[1] - 1])
+
+    return window_sum
+
+
+def _window_sum_3d(image, window_shape):
+
+    window_sum = _window_sum_2d(image, window_shape)
+
+    window_sum = da.cumsum(window_sum, axis=2)
+    window_sum = (window_sum[:, :, window_shape[2]:-1]
+                  - window_sum[:, :, :-window_shape[2] - 1])
+
+    return window_sum
```

Comment:
I think this can be made n-dimensional without too much difficulty...

```python
integral_image = image
for ax in range(image.ndim):
    integral_image = da.cumsum(integral_image, axis=ax)
window_sums = integral_image
for ax in range(image.ndim):
    start = [slice(None),] * image.ndim
    end = [slice(None),] * image.ndim
    start[ax] = slice(0, -window_shape[ax])
    end[ax] = slice(window_shape[ax], None)
    # (not sure why the -1 margins were necessary, btw?)
    window_sums = window_sums[end] - window_sums[start]
```

Not sure if that's exactly right but should be close...!

Reply:
Ah thank you for the suggestion! I'll work on this soon